src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -78,35 +82,42 @@ private void clearCommand() {
 		updateCurrentCommandField();
 	}
 
+    private void runCommand() {
+		commandManager.runCommand();
+    }
+
 	public void updateCurrentCommandField() {
 		currentCommandField.setText(commandManager.getCurrentCommandString());
 	}
 
-	public void deleteObservers() {
-		commandManager.getChangePublisher().clearObservers();
-		this.updateObserverListField();
+	public void deleteObservers(ActionEvent e) {
+		JToggleButton button = (JToggleButton) e.getSource();
+		if(button.isSelected()) {
+			commandManager.getChangePublisher().clearObservers();
+			updateObserverListField();
+			button.setText("Reset observers");
+		} else {
+			updateObserverListField();
```

Comment:
What does reset observers do? The idea was to revert observers deleted by 'delete observers'